### PR TITLE
Implement jax.scipy.stats.truncnorm

### PR DESCRIPTION
**Pull request by adrn**
Following from #12618, here is an implementation of the logpdf, logcdf, and logsf (and un-log versions) for `scipy.stats.truncnorm`. Much of this is ported over from scipy itself, but I had to change some logic to use `select()` statements. I was a little confused by the test structure but think I figured out a solution to enforce that the upper limit parameter `b` is always larger than the lower limit `a`, and the evaluating values `x` are between these values.

### CODE REVIEW

**Comment by adrn** [comment]
@jakevdp I don't have privileges to request a review from you, so pinging you instead! (and I just filled out the CLA)

File: tests/scipy_stats_test.py
Comment (jakevdp) [inline tests/scipy_stats_test.py:487]: What happens if these are not in order `a < x < b`? It should return `-inf`, correct? Should we just let the values be out of order so as to test that?

If we need them in order, it might be simpler to do something like this, and then cast to the desired dtype:
```python
a, x, b = rng((3, *shape), dtype).sort(0)
```

**Review by jakevdp — COMMENTED** [review]
Looks great! Just a quick comment on tests.

File: tests/scipy_stats_test.py
Comment (adrn) [reply tests/scipy_stats_test.py:487]: Yea, true, I guess we can relax the need for `x` to be in bounds, but `b` must be greater than `a`. But since `a` or `b` can have different shapes, I couldn't figure out a clean way to use `np.sort()` to get this to work. My hack might be the simplest way of enforcing this, but I could think more about it!

Comment (jakevdp) [reply tests/scipy_stats_test.py:486]: It looks like when `a >= b` the scipy version returns `nan`:
```python
from scipy.stats import truncnorm
print(truncnorm.logpdf(1, a=2, b=0))
# nan
```
I believe `_CheckAgainstNumpy` will check the locations of NaN values, so it might be best in the tests to check the full range of values to ensure that we're matching that behavior.

**Comment by adrn** [comment]
@jakevdp Sorry for the delay here! I removed the hack parameter value enforcement from the tests (now it tests invalid parameter values as well) and rebased because of a conflict.

**Review by jakevdp — COMMENTED** [review]
Thanks for circling back to this - looks great! Two last things before we merge:

- please add the new functionality to the HTML docs in https://github.com/google/jax/blob/main/docs/jax.scipy.rst
- please squash all your changes into a single commit (see [single-change commits](https://jax.readthedocs.io/en/latest/contributing.html#single-change-commits-and-pull-requests))

Thanks!

**Comment by adrn** [comment]
Cool - thanks for the review! I just squashed the commits and added it to the docs.

**Review by jakevdp — APPROVED** [review]

**Comment by adrn** [comment]
Fixed a small docs issue and pushed again, so tests might need to be approved again.

**Comment by jakevdp** [comment]
The doc build is failing in a strange way, that seems to be related to our lazy loading of scipy. I'm not sure what the fix might be...

File: jax/_src/scipy/stats/truncnorm.py
Comment (jakevdp) [inline jax/_src/scipy/stats/truncnorm.py:22]: I think this is the issue: importing `stats` here leads to a circular import.

Using `from jax._src.scipy.stats import norm` should fix this.

Comment (adrn) [reply jax/_src/scipy/stats/truncnorm.py]: Got it - done!

**Comment by jakevdp** [comment]
Thanks for the PR!

**Comment by adrn** [comment]
Thanks for the review!

**Pull request merged**
